controllers/chaosimpl/networkchaos/trafficcontrol/impl.go

```diff
@@ -299,15 +299,20 @@ func (impl *Impl) ApplyTc(ctx context.Context, m *podnetworkchaosmanager.PodNetw
 		}
 		targetPods = append(targetPods, pod)
 	}
-	dstIpset := ipset.BuildIPSet(targetPods, externalCidrs, networkchaos, string(tcType[0:2])+ipSetPostFix, m.Source)
-	impl.Log.Info("apply traffic control with filter", "sources", m.Source, "ipset", dstIpset)
+	dstSetIPSet, dstOtherIPSets := ipset.BuildIPSets(targetPods, externalCidrs, networkchaos, string(tcType[0:2])+ipSetPostFix, m.Source)
```

Comment:
Why should different IPSet types be handled separately here, and can it be handled together?

Reply:
The `list:set` set is the only one that needs to be referenced by iptables, as it stores all other IP sets, but we still need to return all IP sets, because chaos-daemon must create all of them.

If the `BuildIPSets` returned a single array, the caller would need to search for `list:set` type or assume that it's the first element of array. I think using separate return value is more elegant.

Reply:
Yes, I agree with using separate return value is more elegant. So can we use two functions to generate implement `BuildIPSet` logic?

Reply:
I'm not sure I follow. Can you explain in more detail what do you want to change? Right now `BuildIPSets` builds all IP sets definitions, including a `list:set` set that stores all other sets. Would you rather want the callers to assemble all sets type by type? That'd probably result in more code duplication, as it's done in at least two places (tc and partition).

Reply:
We can use `BuildIPSets` to generate `NetIPSet` and `NetPortIPSet`, and then use these to generate `list:set`.

Reply:
I just think `dstOtherIPSets` is confusing and ugly.

Reply:
> Yes, I agree with using separate return value is more elegant. So, can we use two functions to generate implement BuildIPSet logic?

> I just think dstOtherIPSets is confusing and ugly.

Do we really need separate it into two functions? Or only rename `dstOtherIPSets` to `dstIPSets` is good enough. 🥲

Reply:
`dstIPSets` is ok, and separating into two functions is also necessary.

Reply:
I don't know what to do then. Please tell me whether I should revert the change and change the variable name or leave it as is.

Reply:
The current implementation is ok, just keep it.